### PR TITLE
Fix for compnerd/swift-win32/issues/272

### PR DESCRIPTION
**Pull request by yostane**
- Application main loads the delegate name from info.plist
- main: remove delegate param since it's loaded from info.plist
- Calculator: copy info.plist in CMakeLists (calculator exe was not loading before that)

### CODE REVIEW

File: Sources/SwiftWin32/Application/ApplicationMain.swift [hunk 1]
```diff
@@ -43,8 +43,7 @@ private func WaitMessage(_ dwMilliseconds: UINT) -> Bool {
 @discardableResult
 public func ApplicationMain(_ argc: Int32,
                             _ argv: UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>,
-                            _ application: String?,
-                            _ delegate: String?) -> Int32 {
+                            _ application: String?) -> Int32 {
```
Comment (compnerd) [inline Sources/SwiftWin32/Application/ApplicationMain.swift:46]: I would prefer to retain the ability to explicitly specify the delegate if the user so desires.  I don't think that it really adds much in terms of complexity.  If one is not specified, we can fallback to the `Info.plist` value, and if one is not present, we can default the delegate.

File: Sources/SwiftWin32/Application/ApplicationMain.swift [hunk 2]
```diff
@@ -58,8 +57,17 @@ public func ApplicationMain(_ argc: Int32,
...
+  if let path = Bundle.main.path(forResource: "Info", ofType: "plist") {
+      if let contents = FileManager.default.contents(atPath: path) {
+        Application.shared.information =
+            try? PropertyListDecoder().decode(Application.Information.self,
+                                              from: contents)
+      }
+  }
+
+  // Setup ApplicationDelegate using the 'SceneDelegateClassName' info from the plist
+  if let delegate = Application.shared.information?.scene?.configurations?[SceneSession.Role.windowApplication.rawValue]?.first?.delegate {
```
Comment (compnerd) [inline Sources/SwiftWin32/Application/ApplicationMain.swift:70]: The scene delegate and application delegates are separate (or can be).  I don't follow the reasoning for initializing the application delegate from the scene delegate.

**Review by compnerd — CHANGES_REQUESTED** [review]
Thanks for the change!  Would you mind splitting out the CMake change for the `Info.plist` copying to a separate change?

**Comment by yostane** [comment]
> Thanks for the change! Would you mind splitting out the CMake change for the `Info.plist` copying to a separate change?

Ok, I'll close this one and make separate PRs

**Pull request closed without merging**